tests/test_permissions.py

```diff
@@ -0,0 +1,451 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015-2019 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Tests for Permission class."""
+
+import pytest
+from flask_principal import ActionNeed, Need, RoleNeed, UserNeed
+from invenio_accounts.models import Role, User
+from invenio_db import db
+
+from invenio_access.models import ActionRoles, ActionSystemRoles, ActionUsers
+from invenio_access.permissions import (
+    ParameterizedActionNeed,
+    Permission,
+    any_user,
+    authenticated_user,
+    superuser_access,
+)
+
+# Test cases summary
+
+# Needs to be tested:
+
+# UserNeed
+# RoleNeed
+# ActionNeed
+# ParameterizedActionNeed
+# SystemRoleNeed
+
+# Test cases:
+
+# 1. allow UserNeed
+# 2. deny UserNeed
+# 3. allow RoleNeed
+# 4. deny RoleNeed
+# 5. allow UserNeed, RoleNeed
+# 6. deny UserNeed, RoleNeed
+# 7. allow ActionNeed -> UserNeed
```

Comment:
Minor: On the top it can be specified that `->` means expansion. (Cuz it means expansion, right?)